backend/routes/endSession.js

```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+const router = express.Router();
+const admin = require('firebase-admin');
+// const advanceQueue = require('./advanceQueue'); // Import advanceQueue function
+
+// End Session Endpoint
+router.post('/endSession', async (req, res) => {
+    try {
+        // Extract location and user UID from the request body
+        const { location, uid } = req.body
+        if (!location || !uid) {
+            return res.status(400).json({ message: 'Location and UID are required.' });
+        }
+
+        // Check if the location document exists
+        const locationDoc = await admin.firestore()
+        .collection('locations')
+        .doc(location)
+        .get();
+
+        if (!locationDoc.exists) {
+        return res.status(404).json({ message: 'Location not found.' });
+        }
```

Comment:
This will cost an extra read to Firebase on each call to the endpoint. Instead, check the result of the active players ref document because that must be called regardless and it will check for a valid location...